pyrenew_covid_wastewater/hosp_only_ww_right_truncation_model.py

```diff
@@ -0,0 +1,225 @@
+        )
+
+        rtu = repeat_until_n(
+            data=jnp.exp(log_rtu_weekly),
+            n_timepoints=n_datapoints,
+            offset=0,
+            period_size=7,
+        )
+
+        generation_interval_pmf = self.generation_interval_pmf_rv()
+
+        inf_with_feedback_proc_sample = self.inf_with_feedback_proc(
+            Rt=rtu,
+            I0=i0,
+            gen_int=generation_interval_pmf,
+        )
+
+        latent_infections = jnp.concat(
+            [
+                i0,
+                inf_with_feedback_proc_sample.post_initialization_infections,
+            ]
+        )
+        numpyro.deterministic("rtu", rtu)
+        numpyro.deterministic("rt", inf_with_feedback_proc_sample.rt)
+        numpyro.deterministic("latent_infections", latent_infections)
+
+        p_hosp_mean = self.p_hosp_mean_rv()
+        p_hosp_w_sd = self.p_hosp_w_sd_rv()
+        autoreg_p_hosp = self.autoreg_p_hosp_rv()
+
+        p_hosp_ar_init_rv = DistributionalVariable(
+            "p_hosp_ar_init",
+            dist.Normal(
+                0,
+                p_hosp_w_sd / jnp.sqrt(1 - jnp.pow(autoreg_p_hosp, 2)),
+            ),
+        )
+        p_hosp_ar_init = p_hosp_ar_init_rv()
+
+        p_hosp_ar = self.p_hosp_ar_proc(
+            n=n_weeks_post_init,
+            autoreg=autoreg_p_hosp,
+            init_vals=p_hosp_ar_init,
+            noise_sd=p_hosp_w_sd,
+        )
+
+        ihr = jnp.repeat(
+            transformation.SigmoidTransform()(p_hosp_ar + p_hosp_mean),
+            repeats=7,
+        )[:n_datapoints]
+        # this is only applied after the hospital_admissions are generated, not to all the latent infections. This is why we cannot apply the ihr in compute_delay_ascertained_incidence
+        # see https://github.com/CDCgov/ww-inference-model/issues/43
+
+        numpyro.deterministic("ihr", ihr)
+
+        hosp_wday_effect_raw = self.hosp_wday_effect_rv()
+        hosp_wday_effect = tile_until_n(hosp_wday_effect_raw, n_datapoints)
+
+        inf_to_hosp = self.inf_to_hosp_rv()
+
+        potential_latent_hospital_admissions = (
+            compute_delay_ascertained_incidence(
+                p_observed_given_incident=1,
+                latent_incidence=latent_infections,
+                delay_incidence_to_observation_pmf=inf_to_hosp,
+            )[-n_datapoints:]
+        )
+
+        if right_truncation_offset is not None:
+            right_truncation_multiplier_tail = jnp.flip(
+                self.right_truncation_cdf_rv()[right_truncation_offset:]
+            )
+            n_points_to_prepend = (
+                n_datapoints - right_truncation_multiplier_tail.shape[0]
+            )
+            right_truncation_multiplier = jnp.pad(
+                right_truncation_multiplier_tail,
+                (n_points_to_prepend, 0),
+                mode="constant",
+                constant_values=(1, 0),
+            )
+        else:
+            right_truncation_multiplier = 1
```

Comment:
1. Name the variable `fraction_already_reported` or similar rather than `right_truncation_multiplier`?
2. Might it be more readable to construct it by creating a `truncated` mask and then doing
```python
fraction_already_reported = jnp.where(
    truncated,
    right_truncation_multiplier_tail
    1)
```
?

Reply:
I don't think this works because `truncated` is longer than `right_truncation_multiplier_tail`?

Reply:
You're totally right, apologies. Do think the variable name suggestion makes sense.

Reply:
Changed the name in https://github.com/CDCgov/pyrenew-covid-wastewater/pull/17/commits/87186c088d6a20b2648587d790173db3424d2afc